src/lang/diagnostics/mod.rs

```diff
@@ -143,23 +160,32 @@ impl DiagnosticsControllerThread {
                     error!("caught panic in diagnostics worker");
                 }
             }
+            sender.send(WorkerStatus::Stopped).unwrap();
```

Comment:
I'm afraid of two facts here:
1. This is panicking
2. This is not called from a drop guard, meaning it is possible that some cautionless code change may cause this to never be called.

Reply:
1. Yes but it shouldn't unless we do something wrong 
2. It will be clear when it happens when we have any diagnostics tests since diagnostics will not be reported

Reply:
Ad 1. Please replace this unwrap with an `expect` stating this should never happen "because"
Ad 2. Not sure I understand, can you rephrase plz?

Reply:
This is no longer relevant after rewrite of logic to `TaskHandle`